pkg/sql/logictest/testdata/logic_test/enums

```diff
@@ -410,3 +410,42 @@ CREATE TABLE t1 (
    INDEX i (x ASC),
    FAMILY "primary" (x, rowid)
 )
+
+# Test that the implicit array type has been created, and that we can use it.
+query T
+SELECT ARRAY['hello']::_greeting
+----
```

Comment:
What about a test of `array['hello'::greeting]` or is that tested elsewhere? It'd be good to test that you can't have a heterogeneous enum array as well.